clippy_lints/src/utils/conf.rs

```diff
@@ -227,22 +229,22 @@ pub fn read(path: Option<&path::Path>) -> (Conf, Vec<Error>) {
 
     assert!(ERRORS.lock().expect("no threading -> mutex always safe").is_empty());
     match toml::from_str(&file) {
-        Ok(toml) => (
-            toml,
-            ERRORS.lock().expect("no threading -> mutex always safe").split_off(0),
-        ),
-        Err(e) => {
+        Ok(toml) => {
             let mut errors = ERRORS.lock().expect("no threading -> mutex always safe").split_off(0);
-            let err_as_str = e.to_string();
-
-            if err_as_str.find("cyclomatic_complexity_threshold").is_some() {
-                errors.push(Error::Toml(
-                    "found deprecated field `cyclomatic_complexity_threshold`. Please use `cognitive_complexity_threshold` instead.".to_string(),
-                ));
-            } else {
-                errors.push(Error::Toml(err_as_str));
+
+            let cyc_field: u64 = (toml as Conf).cyclomatic_complexity_threshold;
```

Comment:
I added this because the compiler didn't let me do `toml.(...)` directly, apparently it needed a type annotation? The way I just did it is probably not what it needs, tho. If this commit works, then I'll fix the type annotation later.

Reply:
Update: did not work.